dtbase/backend/api/sensor/routes.py

```diff
@@ -97,3 +161,95 @@ def list_sensors_of_a_type(type_name):
     """List sensors of a particular type in the database."""
     result = sensors.list_sensors(type_name=type_name, session=db.session)
     return jsonify(result), 200
+
+
+@blueprint.route("/list_sensor_types", methods=["GET"])
+# @login_required
+def list_sensor_types():
+    """List sensor types in the database."""
+    result = sensors.list_sensor_types(session=db.session)
+    return jsonify(result), 200
+
+
+@blueprint.route("/list_measures", methods=["GET"])
+# @login_required
+def list_sensor_measures():
+    """List sensor measures in the database."""
+    result = sensors.list_sensor_measures(session=db.session)
+    return jsonify(result), 200
+
+
+@blueprint.route("/sensor_readings", methods=["POST"])
+# @login_required
+def get_sensor_readings():
+    """
+    Get sensor readings for a specific measure and sensor between two dates.
+
+    POST request should have JSON data (mimetype "application/json") containing:
+        measure_name: Name of the sensor measure to get readings for.
+        sensor_uniq_id: Unique identifier for the sensor to get readings for.
+        dt_from: Datetime string for earliest readings to get. Inclusive. In ISO 8601 format: '%Y-%m-%dT%H:%M:%S'.
+        dt_to: Datetime string for last readings to get. Inclusive. In ISO 8601 format: '%Y-%m-%dT%H:%M:%S'.
+    """
+
+    payload = json.loads(request.get_json())
+
+    required_keys = ["measure_name", "sensor_uniq_id", "dt_from", "dt_to"]
```

Comment:
Optional, but since this keeps coming up, maybe there could be a function in `utils.py` that does a check like this? Takes a dictionary and a set of keys that should be in it, returns something like a set of keys that are missing, or the first key it encounters that is missing. Then this could be replaced with something like
```
for k in utils.missing_keys(payload, required_keys)
    return utils.missing_key_json_error(k, "get_sensor_readings"), 400
```

Reply:
Good idea, outsourced this to utils.py.